ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/LimitPayloadSize.java

```diff
@@ -25,15 +32,21 @@ public static LimitPayloadSize toMB(int mb) {
   }
 
   @Override
-  protected PubsubMessage processElement(PubsubMessage message) throws Exception {
-    message = PubsubConstraints.ensureNonNull(message);
-    int numBytes = message.getPayload().length;
-    if (numBytes > maxBytes) {
-      countPayloadTooLarge.inc();
-      throw new PayloadTooLargeException("Message payload is " + numBytes + "bytes, larger than the"
-          + " configured limit of " + maxBytes);
-    }
-    return message;
+  public Result<PCollection<PubsubMessage>, PubsubMessage> expand(
+      PCollection<PubsubMessage> elements) {
+    return elements.apply(
+        MapElements.into(TypeDescriptor.of(PubsubMessage.class)).via((PubsubMessage message) -> {
+          message = PubsubConstraints.ensureNonNull(message);
+          int numBytes = message.getPayload().length;
+          if (numBytes > maxBytes) {
+            countPayloadTooLarge.inc();
+            throw new RuntimeException(new PayloadTooLargeException("Message payload is " + numBytes
```

Comment:
It's only possible to throw `RuntimeExceptions`. Not sure if there is a better way than to just wrap all other exception types into `RuntimeExceptions`?

Reply:
I think it's lambdas that can't have runtime exceptions. I think we could make turn this into a `new ProcessFunction {...}` and we'd be able to throw `PayloadTooLargeException` here.

I'm also uncertain as to the value we're getting from having our custom exception classes be checked, since it does make us have to deal with additional complications like this. Checked exceptions are nice in theory, though I believe much of the Java community considers checked exceptions a failed experiment. I don't have great references for that, though.

Reply:
We discussed with @relud and he supports abandoning checked exceptions, and making all of our custom exception types be runtime.